csrc/preseg_passes/propagate_shardings.cpp

```diff
@@ -0,0 +1,47 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <preseg_passes/propagate_shardings.h>
+
+#include <vector>
+
+#include <ir/interface_nodes.h>
+#include <ir/utils.h>
+#include <multidevice/utils.h>
+
+namespace nvfuser::preseg_passes {
+
+void PropagateShardingsPass::runPass(Fusion* fusion) {
+  for (Expr* expr : fusion->exprs()) {
```

Comment:
I replaced some `auto`s with explicit types; otherwise, the content is identical to the old propagateShardings function.